ChronoVisor/include/Story.h

```diff
@@ -54,16 +62,30 @@ class Story {
     void setName(const std::string &name) { name_ = name; }
     void setSid(uint64_t sid) { sid_ = sid; }
     void setCid(uint64_t cid) { cid_ = cid; }
-    void setStats(const StoryStats &stats) { stats_ = stats; }
+    void setStats(const StoryStats &stats) { std::lock_guard<std::mutex> lock(statsMutex_); stats_ = stats; }
```

Comment:
What is this mutex for? Since stats is not a global variable, and story objects are accessed only through their parent chronicle objects (protected by mutexes), do you need to add this lock here?